### PR TITLE
Add an expected failng test for #14086

### PR DESCRIPTION
**Pull request by bhack**
This is a tentative export for an expecting failing test for https://github.com/keras-team/keras/issues/14086.

But the test seems to pass locally.

Test is base on @ogrisel minimal example at https://github.com/keras-team/keras/issues/14086#issuecomment-784038944

### CODE REVIEW

File: keras/engine/training_test.py [hunk 1]
```diff
@@ -1427,6 +1428,30 @@ def test_outputs_are_floats(self):
     self.assertIsInstance(loss, float)
     self.assertIsInstance(accuracy, float)
 
+  @keras_parameterized.run_all_keras_modes(always_skip_v1=True)
+  @unittest.expectedFailure
```
Comment (qlzh727) [inline keras/engine/training_test.py:1432]: Is this line expected here?

**Review by qlzh727 — COMMENTED** [review]
Thanks for sending this PR. I assume this is an minimal repro for the issue and not intended for merge, is it?

**Comment by bhack** [comment]
> Thanks for sending this PR. I assume this is an minimal repro for the issue and not intended for merge, is it?
We have started to experimentally review/merge some real expecting failing in TF test like:
https://github.com/tensorflow/tensorflow/pull/51538
https://github.com/tensorflow/tensorflow/pull/51374

We have an internal draft on this so if you are interested please contact me privately.
But just to summarize it is mainly is for:
* exactly reproduce a confirmed bug (or fr?) with a minimal code that is almost near to our test expected QA
* not continuously ping over time old tickets with "is it solved in the last release? or is it solved nightly?)
* host a clear coverage of what a contributor bugfix PR will need to cover/pass
* eventually use a specific Github label for issues having an "expecting failing test" coverage

So, If this is really going to fail as the issue and forum claims it is supposed to be merge and it will fail when it is solved with the internal commit activities or with a bugfix PR in the future. 

But for me it isn't failing locally with master/nightly and so probably test will passing also in the CI. 
In that case  do we need to modify the test for a better coverage of the issue claim or is it already solved on master/nightly and we could close this?

File: keras/engine/training_test.py (hunk 1 above)
Comment (bhack) [reply keras/engine/training_test.py:1432]: It is expected, an expected failing test will need to fail only when it will be solved.

**Comment by bhack** [comment]
Ok now I can reproduce the issue.

File: keras/engine/training_test.py [hunk 2]
```diff
@@ -1427,6 +1428,34 @@ def test_outputs_are_floats(self):
...
+      loss[i], accurancy[i] = model.test_on_batch(x[:batch_size], 
+                                                  y[:batch_size])
+    model.evaluate(x,y, batch_size=batch_size, verbose=0)
+    for i in range(3):
+      loss_1[i], accurancy_1[i] = model.test_on_batch(x[:batch_size], 
+                                                      y[:batch_size])
+    self.assertAllEqual(loss, loss_1, 
+                        "https://github.com/keras-team/keras/issues/14086")
+    self.assertAllEqual(accurancy, accurancy_1, 
+                        "https://github.com/keras-team/keras/issues/14086")
```
Comment (ogrisel) [inline keras/engine/training_test.py:1457]: typo: `accurancy` => `accuracy`, `accurancy_1` => `accuracy_1`

File: keras/engine/training_test.py [hunk 3]
```diff
@@ -2929,6 +2930,26 @@ def test_metric_state_reset_between_fit_and_evaluate(self):
...
+        loss='mae',
+        metrics=[acc_obj],
+        optimizer=RMSPropOptimizer(learning_rate=0.001),
+        run_eagerly=testing_utils.should_run_eagerly())
+
+    x_test = np.random.random((10, 4))
+    y_test = np.random.random((10, 1))
+    loss, acc = model.test_on_batch(x_test[:2],y_test[:2])
+    model.evaluate(x_test, y_test)
+    loss_1, acc_1 = model.test_on_batch(x_test[:2],y_test[:2])
```
Comment (qlzh727) [inline keras/engine/training_test.py:2949]: Could u add another model.eval below this line, and make sure the output are the same as the first model.eval()?

This will make sure we don't include the any result from model.test_on_batch().

File: keras/engine/training.py [hunk 4]
```diff
@@ -1855,8 +1857,6 @@ class during training. This can be useful to tell the model to "pay
       self.train_function = self.make_train_function()
       logs = self.train_function(iterator)
 
-    if reset_metrics:
```
Comment (qlzh727) [inline keras/engine/training.py:1858]: I think this is probably needed as well, otherwise the follow up model.eval() will accumulate the metric result from this train/test_on_batch.

**Review by qlzh727 — CHANGES_REQUESTED** [review]
Thanks for the fix. Overall it looks good. We need to make sure train_on_batch doesn't have any side effects left.

File: keras/engine/training_test.py (hunk 3 above)
Comment (bhack) [reply keras/engine/training_test.py:2949]: Done

File: keras/engine/training.py (hunk 4 above)
Comment (bhack) [reply keras/engine/training.py:1858]: See the extended check, it seems not.

Comment (qlzh727) [reply keras/engine/training.py:1858]: ok. Good to know.

**Review by qlzh727 — APPROVED** [review]
